packages/calcite-components/src/utils/form.tsx

```diff
@@ -158,6 +169,51 @@ function hasRegisteredFormComponentParent(
   return hasRegisteredFormComponentParent;
 }
 
+function displayValidationMessage(event: Event) {
+  if (event) {
+    // target is the hidden input, which is slotted in the actual form component
+    const hiddenInput = event?.target as HTMLInputElement;
+
+    // not necessarily a calcite-input, but we don't have an HTMLCalciteFormElement type
+    const formComponent = hiddenInput?.parentElement as HTMLCalciteInputElement;
+
+    const componentTag = formComponent?.nodeName?.toLowerCase();
+    const componentTagParts = componentTag?.split("-");
+
+    if (componentTagParts.length < 2 || componentTagParts[0] !== "calcite") {
+      return;
+    }
+
+    // prevent the browser from showing the native validation popover
+    event?.preventDefault();
```

Comment:
Can you check if this prevents the browser placing focus on the first invalid input (and scrolling to it)?